app/models/piece.rb

```diff
@@ -1,3 +1,64 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
+
+  def check_square(x, y)
```

Comment:
In Ruby, you'll often see `predicate methods`. Just meaning, a method with a `?` at the end that answers true or false.

For this method, to go with that style, I'd rename it to something like... `square_taken?(x, y)` or the reverse `square_empty?`